### ISSUE TITLE
Exception: System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension.

### ISSUE THREAD

**Issue by begrimshaw** [PowerShell/PSReadLine#2553]
I am getting a huge error that I have never seen before. I can't even run a file to print "Hello" anymore. I am using Visual Studio Code and running a c++ file.

**Comment by SpitFire-666** [comment]
I received this error when doing a Ctrl+R (history) search:

```
Oops, something went wrong.  Please report this bug with the details below.                                                                     
Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new                                                                               
-----------------------------------------------------------------------                                                                         
Last 13 Keys:                                                                                                                                   
 d a t e Enter                                                                                                                                  
 DownArrow Ctrl+r r o b p o Escape                                                                                                              
                                                                                                                                                
Exception:                                                                                                                                      
System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension.  
Parameter name: top                                                                                                                             
Actual value was -15.                                                                                                                           
   at System.Console.SetCursorPosition(Int32 left, Int32 top)                                                                                   
   at Microsoft.PowerShell.PSConsoleReadLine.ReallyRender(RenderData renderData, String defaultColor)                                           
   at Microsoft.PowerShell.PSConsoleReadLine.ForceRender()                                                                                      
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()                                                                                        
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)                                     
-----------------------------------------------------------------------                                                                         
```

**Comment by jarvis-1805** [comment]
I too received this bug while executing a python script, but was able to execute the script after this bug has shown up.
```
Oops, something went wrong.  Please report this bug with the details below.
Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new
-----------------------------------------------------------------------
Last 19 Keys:
 UpArrow Enter
 Ctrl+l UpArrow Enter
 Ctrl+l UpArrow Enter
 UpArrow Enter
 UpArrow Enter
 UpArrow Enter
 UpArrow Enter
 UpArrow Enter
 Ctrl+l

Exception:
System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension.
Parameter name: top
Actual value was -7.
   at System.Console.SetCursorPosition(Int32 left, Int32 top)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)
-----------------------------------------------------------------------
```

**Comment by FelipeCoimbra** [comment]
I get this after hitting Tab for autocompletion with the Complete handler
```ps
Set-PSReadlineKeyHandler -Key Tab -Function Complete
```
```ps
PS C:\Users\t-felipev> set-
Display all 215 possibilities? (y or n) y
Oops, something went wrong.  Please report this bug with the details below.
Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new
-----------------------------------------------------------------------
Last 200 Keys:
 - M o d u Tab Space - Backspace P S R e a d L i n e Space - A l l o Tab Backspace Backspace Backspace Backspace Backspa
Oops, something went wrong.  Please report this bug with the details below.
Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new
-----------------------------------------------------------------------
Last 200 Keys:
 - Tab Backspace Backspace Backspace P S R Tab Space - K e y Space T a b - F u n c t i o n Space Ctrl+LeftArrow / Backsp
Oops, something went wrong.  Please report this bug with the details below.
Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new
-----------------------------------------------------------------------
Last 200 Keys:
 - K e y Space T a b - F u n c t i o n Space Ctrl+LeftArrow / Backspace LeftArrow Space Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow C o p Backspace m p l Tab Enter
 UpArrow ' Backspace @ Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+LeftArrow Ctrl+` ` Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+RightArrow Ctrl+` Space Ctrl+Backspace Ctrl+Backspace Ctrl+Backspace Ctrl+Backspace Ctrl+Backspace Ctrl+Backspace Ctrl+Backspace Ctrl+Backspace Ctrl+Backspace Ctrl+Backspace Ctrl+Backspace Ctrl+Backspace Ctrl+Backspace a b Tab Tab Tab Tab Backspace Tab Backspace S e Tab t Tab Tab y Ctrl+Backspace Ctrl+l s t Backspace e t - Tab Tab y Ctrl+Backspace UpArrow Ctrl+Backspace M e n u C o m p l e t e Enter
 s e t - Tab y Tab Tab Tab n a c Tab Tab Tab Tab Tab Backspace Backspace Tab DownArrow RightArrow RightArrow DownArrow LeftArrow LeftArrow DownArrow DownArrow DownArrow RightArrow RightArrow UpArrow UpArrow UpArrow UpArrow UpArrow UpArrow UpArrow UpArrow UpArrow UpArrow Ctrl+Backspace Ctrl+Backspace Ctrl+RightArrow Ctrl+Backspace Ctrl+Backspace Ctrl+l s Ctrl+l Ctrl+l Ctrl+l Ctrl+l s Backspace e t - Tab Tab Tab y UpArrow UpArrow UpArrow DownArrow DownArrow DownArrow DownArrow Ctrl+l Tab y

Exception:
System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension.
Parameter name: top
Actual value was -80.
   at System.Console.SetCursorPosition(Int32 left, Int32 top)
   at Microsoft.PowerShell.PSConsoleReadLine.Menu.EnsureMenuAndInputIsVisible(IConsole console, Int32 tooltipLineCount)
   at Microsoft.PowerShell.PSConsoleReadLine.Menu.DrawMenu(Menu previousMenu)
   at Microsoft.PowerShell.PSConsoleReadLine.PossibleCompletionsImpl(CommandCompletion completions, Boolean menuSelect)
   at Microsoft.PowerShell.PSConsoleReadLine.CompleteImpl(Boolean menuSelect)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)
-----------------------------------------------------------------------
```

**Comment by daxian-dbw** [comment]
This issue was already fixed (see #1306). Please upgrade to the [2.1.0 version of PSReadLine](https://www.powershellgallery.com/packages/PSReadLine/2.1.0) from PowerShell Gallery.
See the [upgrading section](https://github.com/PowerShell/PSReadLine#upgrading) for instructions. Please let us know if you run into the same issue with the latest version.
